types/lodash/lodash-tests.ts

```diff
@@ -2337,7 +2338,7 @@ _.chain([1, 2, 3, 4]).unshift(5, 6); // $ExpectType LoDashExplicitWrapper<number
         collection; // $ExpectType NumericDictionary<AbcObject>
     });
 
-    fp.forEach(stringIterator, ""); // $ExpectType string
+    fp.forEach(stringIterator, ""); // $ExpectType ArrayLike<string>
```

Comment:
Yeah, this is a little dubious. Not the worst thing, though, so maybe we just need to live with it.

I had one thought - instead of completely ignoring `StringIterator`, we replace it with `ListIterator<T>`? Then `generate-fp` should be smart enough to merge this overload with the other  `ListIterator<T>` overload(s), but still handle `forEach`/`forEachRight` correctly. Then we would get something like `collection: string` for the second argument. Which is great if `T` is `string`, but not so great if `T` is anything else. So maybe not the best idea.

Reply:
I tried this and it did not go well. Maybe I did it wrong, but the tests were complaining about missing types and it seemed... bad.

Reply:
Yeah I was worried it wouldn't work well. Let's leave it as you have it then.